src/layouts/sidebar-sidecar/utils/prepare-nav-data-for-client.ts

```diff
@@ -69,10 +70,32 @@ function prepareNavNodeForClient(node: NavNode, basePaths: string[]): MenuItem {
       id: slugify(`menu-item-${node.path}`, { lower: true }),
     }
   } else if (isNavDirectLink(node)) {
-    // For direct link nodes, add `id` only
-    return {
-      ...node,
-      id: slugify(`external-url-${node.title}`, { lower: true }),
+    const id = slugify(`external-url-${node.title}`, { lower: true })
+    // Check if there is data that disagrees with DevDot's assumptions.
+    // This can happen because in the context of dot-io domains,
+    // authors may write NavDirectLinks with href values that are
+    // internal to the site, but outside the current docs route.
+    // For example, an author working in the Consul sidebar may
+    // create a NavDirectLink with an href of "/downloads".
+    // Here in DevDot, we want this URL to be "/consul/downloads",
+    // and we want to use an internal rather than external link.
+    const hrefIsNotAbsolute = !isAbsoluteUrl(node.href)
+    if (hrefIsNotAbsolute) {
+      // If we have a non-absolute NavDirectLink,
+      // convert it to a NavLeaf node, and treat it similarly.
+      // Note that the `fullPath` added here differs from typical
+      // NavLeaf treatment, as we only use the first part of the `basePath`.
+      // (We expect this to be the product slug, eg "consul").
+      return {
+        title: node.title,
+        path: node.href,
+        id,
+        fullPath: `/${basePaths[0]}${node.href}`,
+      }
```

Comment:
I think instead of setting every property (like `title`) we want to spread `node`, "delete" `href` (by setting it to `null`), and only add the properties we're using to correct the node. That way if we ever want to add more properties to nodes we won't have tricky bugs coming from not spreading in this block. Other proposed change is to alphabetize the properties we're setting.

```suggestion
      return {
        ...node,
        fullPath: `/${basePaths[0]}${node.href}`,
        href: null,
        id,
        path: node.href,
      }
```

Reply:
Sounds good, applied!

For context, had switched this because it felt like we're trying to switch the node structure here from `NavDirectLink` to `NavLeaf`. It made me think we might even want to do this transform as a separate, pre-cursor step to the transformations we have set up in `prepare-nav-data-for-client`. Otherwise, eg if we do add properties to `NavDirectLink`, and spread all of them, we could end up with a weird hybrid between the two node types (which I think could also cause unexpected bugs).